packages/sdk/src/agent.ts

```diff
@@ -0,0 +1,171 @@
+import { AskarModule } from '@credo-ts/askar'
+import {
+  Agent,
+  DidsModule,
+  KeyDerivationMethod,
+  type ModulesMap,
+  X509Module,
+  type X509ModuleConfigOptions,
+} from '@credo-ts/core'
+import {
+  ConnectionsModule,
+  CredentialsModule,
+  DidCommModule,
+  DiscoverFeaturesModule,
+  HttpOutboundTransport,
+  MediationRecipientModule,
+  MessagePickupModule,
+  OutOfBandModule,
+  ProofsModule,
+  WsOutboundTransport,
+} from '@credo-ts/didcomm'
+import { OpenId4VcHolderModule } from '@credo-ts/openid4vc'
+import { agentDependencies } from '@credo-ts/react-native'
+import { askar } from '@openwallet-foundation/askar-react-native'
+import { type LogLevel, logger } from './logger'
+import type { RequiredFields, StripOptionalAndUndefined } from './types'
+
+export type InitializeAgentOptions<AdditionalModules extends ModulesMap = ModulesMap> = {
+  /**
+   *
+   * Label used when using DIDComm to convey who you are
+   *
+   */
+  label: string
+
+  /**
+   *
+   * Unique identifier of your wallet storage
+   *
+   */
+  id: string
+
+  /**
+   *
+   * Key that will be used to securely open your wallet
+   *
+   * @security
+   * This property must dealt with in great care, such as receiving it from a biometrics check
```

Comment:
Ideally we export utils for this from the SDK at some point as well. Basically what we have now in the secure unlock provider

Reply:
Yeah, thats the plan